mono/metadata/unity-memory-info.c

```diff
@@ -638,6 +614,15 @@ static void CollectMonoImageFromAssembly(MonoAssembly *assembly, void *user_data
 
 MonoManagedMemorySnapshot* mono_unity_capture_memory_snapshot()
 {
+	int wasDisabled = GC_is_disabled();
```

Comment:
I think this is ref counted internally, so I am not sure you need the if checks

Reply:
fixed up in current version